### PR TITLE
 Adding support to import submodules

### PR DESCRIPTION
**Pull request by dhawalseth**
Adding support to import submodules

### CODE REVIEW

**Comment by aledpardo** [comment]
Hi @samcontesse !

Do you plan to accept this PR?

**Comment by aledpardo** [comment]
Hey @samcontesse how could I write changes to this PR? Sorry for silly question :)

File: in/cmd/main.go [hunk 1]
```diff
@@ -54,6 +54,15 @@ func main() {
 	execGitCommand([]string{"remote", "update"})
 	execGitCommand([]string{"merge", "--no-ff", "--no-commit", mr.SHA})
 
+	// After merging, clone submodules if git submodules file is present
+	if request.Params.Submodules != "none" && fileExists(".gitmodules") {
+        var source_str = "git@"+request.Source.GetGitlabHost()+":"
+        var target_str = "https:\\/\\/gitlab-ci-token:"+request.Source.PrivateToken+"@"+request.Source.GetGitlabHost()+"\\/"
+
+        execSedCommand([]string{"-i", "s/"+source_str+"/"+target_str+"/g", ".gitmodules"})
```
[the remark this answers is not in the document]
Comment (aledpardo) [reply in/cmd/main.go:62]: Or alternatively, we could pass the `--recurse-submodules` to the `clone` command? That would execute the modules initialization in the same command scope as the clone, which already runs authenticated.

File: in/cmd/main.go [hunk 2]
```diff
@@ -54,6 +54,15 @@ func main() {
 	execGitCommand([]string{"remote", "update"})
 	execGitCommand([]string{"merge", "--no-ff", "--no-commit", mr.SHA})
 
+	// After merging, clone submodules if git submodules file is present
+	if request.Params.Submodules != "none" && fileExists(".gitmodules") {
+        var source_str = "git@"+request.Source.GetGitlabHost()+":"
```
[the remark this answers is not in the document]
Comment (aledpardo) [reply in/cmd/main.go:59]: I'm reflecting on the same problem.

I think we can try to solve 3 scenarios:

1. Initialize modules without auth
2. Initialize modules under the same GitLab, using the available token
3. Initialize modules under different servers (GitLab, GitHub, BitBucket) passing specific auth for each server

File: in/cmd/main.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (aledpardo) [reply in/cmd/main.go:62]: My last comment actually doesn't work.

GIT attempts to clone based on the URL contained in `.gitmodules` file.

**Comment by samcontesse** [comment]
Closing due to inactivity. Support for submodules has been added in https://github.com/samcontesse/gitlab-merge-request-resource/pull/89.

**Pull request closed without merging**